ddtrace/internal/writer.py

```diff
@@ -78,6 +86,43 @@ def flush_queue(self):
                 if result_traces_json and 'rate_by_service' in result_traces_json:
                     self._priority_sampler.set_sample_rate_by_service(result_traces_json['rate_by_service'])
 
+        # Dump statistics
+        # NOTE: Do not use the buffering of dogstatsd as it's not thread-safe
+        # https://github.com/DataDog/datadogpy/issues/439
+        if self.dogstatsd:
+            # Statistics about the queue length, size and number of spans
+            self.dogstatsd.gauge('datadog.tracer.queue.max_length', self._trace_queue.maxsize)
+            self.dogstatsd.gauge('datadog.tracer.queue.length', traces_queue_length)
+            self.dogstatsd.gauge('datadog.tracer.queue.size', traces_queue_size)
+            self.dogstatsd.gauge('datadog.tracer.queue.spans', traces_queue_spans)
+
+            # Statistics about the rate at which spans are inserted in the queue
+            dropped, enqueued, enqueued_lengths, enqueued_size = self._trace_queue.reset_stats()
+            self.dogstatsd.increment('datadog.tracer.queue.dropped', dropped)
+            self.dogstatsd.increment('datadog.tracer.queue.accepted', enqueued)
+            self.dogstatsd.increment('datadog.tracer.queue.accepted_lengths', enqueued_lengths)
+            self.dogstatsd.increment('datadog.tracer.queue.accepted_size', enqueued_size)
+
+            # Statistics about the filtering
+            self.dogstatsd.increment('datadog.tracer.traces.filtered', traces_filtered)
+
+            # Statistics about API
+            self.dogstatsd.increment('datadog.tracer.api.requests', len(traces_responses))
+            self.dogstatsd.increment('datadog.tracer.api.errors',
```

Comment:
nice, I like this breakdown.

So we should get: `api.errors + api.responses == api.requests`